tests/test_adaptor.hh

```diff
@@ -95,92 +93,96 @@ namespace rascal {
-      : MultipleStructureManagerCenterFixture<StructureManager, BaseFixture> {
-    using Parent =
-        MultipleStructureManagerCenterFixture<StructureManager, BaseFixture>;
-    using Manager_P_t = typename Parent::Manager_t;
-    using Manager_t = AdaptorNeighbourList<Manager_P_t>;
-    using Manager_T_t = AdaptorStrict<Manager_t>;
-
-    MultipleStructureManagerNLFixture() : Parent{} {
-      for (Manager_P_t & manager : this->managers_center) {
-        for (double cutoff : this->cutoffs) {
-          this->managers_pair.emplace_back(manager, cutoff);
-          this->managers_pair.back().update();
+  struct MultipleStructureManagerNLStrictFixture {
+    using Factory_t =
+        std::tuple<std::tuple<std::string>, std::tuple<double, bool, bool>>;
+    using ManagerTypeHolder_t =
+        StructureManagerTypeHolder<StructureManagerCenters,
+                                   AdaptorNeighbourList, AdaptorStrict>;
+
+    MultipleStructureManagerNLStrictFixture() {
+      for (auto && filename : this->filenames) {
+        for (auto && cutoff : this->cutoffs) {
+          auto a0{std::make_tuple(filename)};
+          auto a1{std::make_tuple(cutoff, consider_ghost_neighbours, cutoff)};
+          this->factory_args.emplace_back(a0, a1);
         }
       }
     }
 
-    ~MultipleStructureManagerNLFixture() = default;
+    ~MultipleStructureManagerNLStrictFixture() = default;
+    const bool consider_ghost_neighbours{false};
+    const std::vector<std::string> filenames{
+        "reference_data/CaCrP2O7_mvc-11955_symmetrized.json",
+        "reference_data/simple_cubic_8.json",
+        "reference_data/small_molecule.json"};
+    const std::vector<double> cutoffs{{1., 2., 3.}};
 
-    std::list<Manager_t> managers_pair{};
+    std::vector<Factory_t> factory_args{};
   };
 
-  template <class StructureManager, class BaseFixture>
-  struct MultipleStructureManagerStrictFixture
-      : MultipleStructureManagerNLFixture<StructureManager, BaseFixture> {
-    using Parent =
-        MultipleStructureManagerNLFixture<StructureManager, BaseFixture>;
-    using Manager_P_t = typename Parent::Manager_t;
-    using Manager_t = AdaptorStrict<Manager_P_t>;
-    // using Manager_T_t = AdaptorStrict<Manager_t>;
-
-    MultipleStructureManagerStrictFixture() : Parent{} {
-      for (Manager_P_t & manager : this->managers_pair) {
-        this->managers_strict.emplace_back(manager, manager.get_cutoff());
-        this->managers_strict.back().update();
+  template <class BaseFixture>
+  struct MultipleStructureFixture : BaseFixture {
+    using Parent = BaseFixture;
+    using ManagerTypeList_t = typename Parent::ManagerTypeHolder_t::type_list;
+    using Manager_t = typename Parent::ManagerTypeHolder_t::type;
+    using ManagerPtr_t = std::shared_ptr<Manager_t>;
+
+    MultipleStructureFixture() : Parent{} {
+      for (auto factory_arg : this->factory_args) {
+        auto manager{make_structure_manager_stack_with_tuple_and_typeholder<
```

Comment:
Ok, I am totally convinced by this new version of stacking adaptors on a manager, but I guess, we should streamline the interface a little bit. Is there a huge overhead by defining the most flexible interface - maybe this one here - and use this everywhere.

The workflow is definitively is very intriguing - and I also see the application of the _simpler_ factory functions. I currently cannot project the actual usage. Will we, as developer, supply these defined stacks?
This definitively has to be documented very well, if we keep the different interfaces to "factorization".

Reply:
agreed.